src/main/java/com/dynatrace/openkit/core/BeaconSender.java

```diff
@@ -120,15 +128,23 @@ public boolean isInitialized() {
      * Shutdown the BeaconSender and wait until it's shutdown (at most {@link BeaconSender#SHUTDOWN_TIMEOUT} milliseconds.
      */
     public synchronized void shutdown() {
-
+        if (logger.isDebugEnabled()) {
+            logger.debug("BeaconSender thread request shutdown");
+        }
         context.requestShutdown();
 
         if (beaconSenderThread != null) {
             beaconSenderThread.interrupt();
             try {
                 beaconSenderThread.join(SHUTDOWN_TIMEOUT);
+                if (logger.isDebugEnabled()) {
+                    logger.debug("BeaconSender thread stopped");
+                }
             } catch (InterruptedException e) {
                 Thread.currentThread().interrupt();
+                if (logger.isDebugEnabled()) {
+                    logger.debug("BeaconSender thread interrupted");

```

Comment:
The log is misleading, since not the BeaconSender thread has been interrupted, but rather the one that's calling the shutdown method